packages/main/src/Tokenizer.ts

```diff
@@ -465,35 +476,34 @@ class Tokenizer extends UI5Element {
 		tokens.forEach(token => { token.selected = !tokensAreSelected; });
 	}
 
-	_handleTokenSelection(event, deselectAll = true) {
-		if (event.target.hasAttribute("ui5-token")) {
-			const deselectTokens = deselectAll ? this._tokens : [event.target];
+	_handleTokenSelection(e: KeyboardEvent | MouseEvent, deselectAll = true) {
+		const target = e.target as Token;
+		if (target.hasAttribute("ui5-token")) {
+			const deselectTokens = deselectAll ? this._tokens : [e.target];
 
 			deselectTokens.forEach(token => {
-				if (token !== event.target) {
-					token.selected = false;
+				if (token !== target) {
+					(token as Token).selected = false;
 				}
 			});
 		}
 	}
 
-	_fillClipboard(shortcutName, tokens) {
+	_fillClipboard(shortcutName: string, tokens: Array<Token>) {
```

Comment:
in order to validate this method we can create enum with all possible values (copy/cut/paste) then some of the castings will be redundant

Reply:
Done